tests/integration/targets/aci_cloud_ctx_profile/tasks/main.yml

```diff
@@ -8,316 +8,247 @@
-    username: "{{item.aci_username}}"
-    password: "{{item.aci_password}}"
-    tenant: ansible_test
-    cloud: "{{item.cloud_type}}"
-    name: ctx_profile_2
-    vrf: ctx_profile_vrf_2
-    region: "{{item.region_2}}"
-    primary_cidr: '10.101.0.0/16'
-    description: "add ctx_profile_2"
-    state: present
-  register: nm_add_another_aci_ctx_profile
-  loop: "{{cloud}}"
-
-- name: Verify nm_add_another_aci_ctx_profile
-  assert:
-    that:
-      - item is changed
-      - item.previous == []
-      - item.current[0].cloudCtxProfile.attributes.name == "ctx_profile_2"
-      - item.current[0].cloudCtxProfile.attributes.dn == "uni/tn-ansible_test/ctxprofile-ctx_profile_2"
-      - item.current[0].cloudCtxProfile.children[2].cloudRsToCtx.attributes.tnFvCtxName == "ctx_profile_vrf_2"
-      - item.current[0].cloudCtxProfile.children[0].cloudCidr.attributes.addr == "10.101.0.0/16"
-      - item.current[0].cloudCtxProfile.children[0].cloudCidr.attributes.primary == "yes"
-      - item.current[0].cloudCtxProfile.attributes.descr == "add ctx_profile_2"
-  loop: "{{nm_add_another_aci_ctx_profile.results}}"
-
-- name: Query aci cloud context profile ctx_profile_1 
-  aci_cloud_ctx_profile:
-    <<: *aci_info
-    host: "{{item.aci_hostname}}"
-    username: "{{item.aci_username}}"
-    password: "{{item.aci_password}}"
-    tenant: ansible_test
-    name: ctx_profile_1
-    state: query
-  register: query_aci_cloud_profile_1
-  loop: "{{cloud}}"
-
-- name: Verify query_aci_cloud_profile_1
-  assert:
-    that:
-    - item is not changed
-    - item.current[0].cloudCtxProfile.attributes.name == "ctx_profile_1"
-    - item.current[0].cloudCtxProfile.children | length == 3
-  loop: "{{query_aci_cloud_profile_1.results}}"
-
-- name: Query all aci cloud context profiles
-  aci_cloud_ctx_profile:
-    <<: *aci_info
-    host: "{{item.aci_hostname}}"
-    username: "{{item.aci_username}}"
-    password: "{{item.aci_password}}"
-    tenant: ansible_test
-    state: query
-  register: query_all
-  loop: "{{cloud}}"
-
-- name: Verify query_all
-  assert:
-    that:
-    - item is not changed
-    - item.current | length == 1
-    - item.current.0.fvTenant.children | length == 2
-  loop: "{{query_all.results}}"
-
-- name: Remove aci cloud context profile
-  aci_cloud_ctx_profile:
-    <<: *aci_info
-    host: "{{item.aci_hostname}}"
-    username: "{{item.aci_username}}"
-    password: "{{item.aci_password}}"
-    tenant: ansible_test
-    name: ctx_profile_1
-    state: absent
-  register: rm_ctx_profile
-  loop: "{{cloud}}"
-
-- name: Verify rm_ctx_profile
-  assert:
-    that:
-    - item.current == []
-    - item.previous.0.cloudCtxProfile.attributes.name == "ctx_profile_1"
-  loop: "{{rm_ctx_profile.results}}"
\ No newline at end of file
+- name: Verify only Cloud Sites are in use.
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- block:  # block specifies execution of tasks within, based on conditions
```

Comment:
If we put a name on the block part, you could put the when condition here I think between the name and the block.
Might be better but up to you.